modules/ml/src/main/java/org/apache/ignite/ml/nn/architecture/MLPArchitecture.java

```diff
@@ -0,0 +1,125 @@
+    /**
+     * List of layers architectures.
+     */
+    private final List<LayerArchitecture> layers;
+
+    /**
+     * Construct an MLP architecture.
+     *
+     * @param inputSize Size of input.
+     */
+    public MLPArchitecture(int inputSize) {
+        layers = new ArrayList<>();
+        layers.add(new LayerArchitecture(inputSize));
+    }
+
+    /**
+     * Construct an MLP architecture.
+     *
+     * @param layers List of layers architectures.
+     */
+    private MLPArchitecture(List<LayerArchitecture> layers) {
+        this.layers = layers;
+    }
+
+    /**
+     * Count of layers in MLP.
+     * @return Layers count.
+     */
+    public int layersCount() {
+        return layers.size();
+    }
+
+    /**
+     * Size of input of MLP.
+     * @return Size of input.
+     */
+    public int inputSize() {
+        return layers.get(0).neuronsCount();
+    }
+
+    /**
+     * Size of output of MLP.
+     * @return Size of output.
+     */
+    public int outputSize() {
+        return layers.get(layersCount() - 1).neuronsCount();
+    }
+
+    /**
+     * Constructs new MLP architecture with new layer added on top of all this architecture layers.
+     *
+     * @param neuronsCnt Count of neurons in new layer.
+     * @param hasBias Flag indicating presence of bias in added layer.
+     * @param f Activation function of a new layer.
+     * @return New MLP architecture with new layer added on top of all this architecture layers.
+     */
+    public MLPArchitecture withAddedLayer(int neuronsCnt, boolean hasBias, IgniteDifferentiableDoubleToDoubleFunction f) {
+        ArrayList<LayerArchitecture> newLayers = new ArrayList<>(layers);
+        newLayers.add(new TransformationLayerArchitecture(neuronsCnt, hasBias, f));
+        return new MLPArchitecture(newLayers);
+    }
+
+    /**
+     * Get architecture of layer with given index.
+     *
+     * @param layer Index of layer to get architecture from.
+     * @return Architecture of layer with given index.
+     */
+    public LayerArchitecture layerArchitecture(int layer) {
+        return layers.get(layer);
+    }
+
+    /**
+     * Get architecture of transformation layer (i.e. non-input layer) with given index.
+     *
+     * @param layer Index of layer to get architecture from.
+     * @return Architecture of transformation layer with given index.
+     */
+    public TransformationLayerArchitecture transformationLayerArchitecture(int layer) {
+        return (TransformationLayerArchitecture)layers.get(layer);
+    }
+
+    /**
+     * Creates config describing network where first goes this config and after goes this method's argument.
+     *
+     * @param second Config to add after this config.
+     * @return New combined configuration.
+     */
+    public MLPArchitecture add(MLPArchitecture second) {
+        assert second.inputSize() == outputSize();
+
+        MLPArchitecture res = new MLPArchitecture(inputSize());
+        res.layers.addAll(layers);
```

Comment:
Avoid direct access to the field. Strange usage of private API

Reply:
I use it in other class, but in the same one, what's wrong with it?